Task "Movie Catalogue"

kapcta opened the pull request:
Pagination was added. Controller & Service were created and new method was added into MovieJdbcRepository class

kapcta committed:
Task "Movie Catalogue"
Pagination was added

ILevchugov commented on src/main/java/io/levchugov/petproject/controller/MovieController.java:
```diff
@@ -0,0 +1,27 @@
+    public ResponseEntity<List<Movie>> getMovieList (
+            @RequestParam(value = "chatId") Long chatId,
+            @RequestParam(value = "page") int page,
+            @RequestParam(value = "unitsOnPage") int unitsOnPage
```
```suggestion
            @RequestParam(value = "limit") int limit
```
либо 
```suggestion
            @RequestParam(value = "size") int size
```

короче, а смысл тот же
для апи с пагинацией более привычные названия для такого параметра

ILevchugov commented on src/main/java/io/levchugov/petproject/controller/MovieController.java:
```diff
@@ -0,0 +1,27 @@
+    private final MovieService movieService;
+
+    @GetMapping("/movies")
+    public ResponseEntity<List<Movie>> getMovieList (
```
давай все параметры сделаем не обязательными, в случае отсутствия chatId, поиск просто по всем фильмам, в случае отсутствия параметров пагинации использовать дефолтные значения, типа 0 и 100

kapcta committed:
Task "Movie Catalogue"
Name of method's arguments was changed, types of all parameters were swapped to false required, and findAllSavedMovieInDB was created in MovieJdbcRepository for search of all movie in DB.

kapcta committed:
Task "Movie Catalogue"
Search by movie's title was added --> getMoviesByPagination in MovieServiceImpl was changed + findMovieOfUserByTitle and findMovieByTitle was created in MovieJdbcRepository

kapcta committed:
Task "Movie Catalogue"
MovieServiceImplTest was created and all test were passed successfully.

kapcta committed:
Task "Movie Catalogue"
MovieControllerTest was created and test was passed successfully.

ILevchugov commented on src/test/java/io/levchugov/petproject/controller/MovieControllerTest.java:
```diff
@@ -0,0 +1,49 @@
+               .andDo(print())
+               .andExpect(status().isOk());
+
+       assertEquals(Collections.singletonList(movie), movieService.getMoviesByPagination("Inception", null, 0, 100));
```
Этот ассерт ничего не проверяет, нужно взять результат выполнения mockMvc и сравнивать ожидаемое с ним

ILevchugov commented on src/test/java/io/levchugov/petproject/service/MovieServiceImplTest.java:
```diff
@@ -0,0 +1,116 @@
+    private final List<Movie> listForCheckFirstAndSecondAndFifthCondition = new ArrayList<>();
+    private final List<Movie> listForCheckFourthCondition = new ArrayList<>();
+
+    @BeforeEach()
```
```suggestion
    @BeforeAll()
```

ILevchugov commented on src/test/java/io/levchugov/petproject/service/MovieServiceImplTest.java:
```diff
@@ -0,0 +1,116 @@
+
+    @Test
+    void testFirstCondition() {
+        Mockito.when(movieJdbcRepository
```
Для каждого использования Mockito.when() необходимо проверять выполнение этого метода через Mockito.verify()

ILevchugov commented on src/test/java/io/levchugov/petproject/service/MovieServiceImplTest.java:
```diff
@@ -0,0 +1,116 @@
+
+    @BeforeEach()
+    public void init() {
+        movieService = new MovieServiceImpl(movieJdbcRepository);
```
вместо этого лучше @Autowired использовать, что бы Spring сам собрал бин

kapcta committed:
Task "Movie Catalogue"
Renaming arguments and methods parameters

ILevchugov commented on src/main/java/io/levchugov/petproject/service/MovieServiceImpl.java:
```diff
@@ -0,0 +1,33 @@
+
+    @Override
+    public List<Movie> getMoviesByPagination(String title, Long chatId, int size, int page) {
+        if (chatId == null && title == null) {
```
а  если при этом size = 5 и page = 1?)

kapcta committed:
Task "Movie Catalogue"
Renaming arguments and methods parameters

kapcta committed:
Task "Movie Catalogue"
Renaming arguments and methods parameters
